### PR TITLE
[feat](Nereids): support show constraint command

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

```
show constraints from t1;
+------+-------------+-----------------------------------------+
| Name | Type        | Definition                              |
+------+-------------+-----------------------------------------+
| fk   | FOREIGN KEY | FOREIGN KEY (id) REFERENCES cir.t1 (id) |
| uk   | UNIQUE      | UNIQUE (id)                             |
| pk   | PRIMARY KEY | PRIMARY KEY (id)                        |
+------+-------------+-----------------------------------------+
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

File: regression-test/data/nereids_syntax_p0/constraint.out [hunk 1]
```diff
@@ -0,0 +1,6 @@
+-- This file is automatically generated. You should know what you did if you want to edit this
+-- !show_constrait --
+fk	ForeignKeyConstraint	(id) references regression_test_nereids_syntax_p0.t1 (id)
```
Comment (morrySnow) [inline regression-test/data/nereids_syntax_p0/constraint.out:3]: it is better use full name, such as `FOREIGN KEY`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowConstraintCommand.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * add constraint command
+ */
+public class ShowConstraintCommand extends Command implements NoForward {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowConstraintCommand.java:37]:
```suggestion
public class ShowConstraintsCommand extends Command implements NoForward {
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowConstraintCommand.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        this.nameParts = nameParts;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        TableIf tableIf = RelationUtil.getDbAndTable(
+                RelationUtil.getQualifierName(ctx, nameParts), ctx.getEnv()).value();
+        List<List<String>> res = tableIf.getConstraintMap().entrySet().stream()
+                        .map(e -> Lists.newArrayList(e.getKey(),
+                                e.getValue().getClass().getSimpleName(),
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowConstraintCommand.java:56]: Is it better to normalize the type name to "PRIMARY KEY/UNIQUE KEY/FOREIGN KEY" etc ?

Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowConstraintCommand.java]: has fixed

File: regression-test/suites/nereids_syntax_p0/constraint.groovy [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        "in_memory" = "false",
+        "storage_format" = "V2",
+        "disable_auto_compaction" = "false"
+        );
+    """
+
+    sql """
+    alter table t1 add constraint pk primary key (id)
+    """
+
```
Comment (xzj7019) [inline regression-test/suites/nereids_syntax_p0/constraint.groovy:43]: pls add case to show multiple fk reference for self or other table

**Comment by XieJiann** [comment]
run buildall

**Review by xzj7019 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**